integration/fuse/src/main/java/alluxio/fuse/AlluxioFuseFileSystem.java

```diff
@@ -279,16 +279,22 @@ public int getattr(String path, FileStat stat) {
       stat.st_mtim.tv_sec.set(ctime_sec);
       stat.st_mtim.tv_nsec.set(ctime_nsec);
 
-      // for shell-based group mapping, use the uid and gid of the corresponding user registered in
-      // unix; otherwise use uid and gid of the user running alluxio-fuse
+      // for shell-based group mapping, try to get the uid and gid of file/dir owner.
+      // If the user does not registered in the unix or we use other group mapping service,
+      // uid and gid of the user running alluxio-fuse will be set.
+      long uid = -1;
+      long gid = -1;
       if (mIsShellGroupMapping) {
         String owner = status.getOwner();
-        stat.st_uid.set(AlluxioFuseUtils.getUid(owner));
-        stat.st_gid.set(AlluxioFuseUtils.getGid(owner));
-      } else {
-        stat.st_uid.set(UID);
-        stat.st_gid.set(GID);
+        uid = AlluxioFuseUtils.getUid(owner);
+        gid = AlluxioFuseUtils.getGid(owner);
+      }
+      if (uid == -1 || gid == -1) {
+        uid = UID;
+        gid = GID;
```

Comment:
This could be a security issue if the user expects the group mapping to work but it silently fails. Perhaps we could disable FUSE group mapping by default, and error out if the user tries to map an unknown user/group?

Reply:
This would be a better approach. I will change it! Thanks

Reply:
The Fuse  `mIsShellGroupMapping` checks if the property key  `alluxio.security.group.mapping.class` is `ShellBasedUnixGroupsMapping`.
However, this property key has been used in many places in our code base: `DefaultFileSystemMaster.setAttribute` and others.

I don't think Alluxio Fuse is important enough to change the default value of this property and Alluxio Fuse is only using a really subset of this property (Never use the `GroupMappingService.getGroups`). 

Maybe creating another property key is better. I created a `FUSE_SHELL_GROUP_MAPPING_ENABLED` property key which controls the value of `mIsShellGroupMapping` in `AlluxioFuseFileSystem`.

Thanks!

Reply:
creating another property sgtm. Should we throw an error if the user explicitly sets shell mapping but `getUid/getGid` fails?

Reply:
If I have a lot of files in the under filesystem, some of them are written by me(user, group registered), some of them are written by others(cannot get uid/gid). I don't want the `ls` command to fail and show nothing in this case.

I would rather let `ls` show the current system user(like my new approach) or show nouser/nogroup(like in previous Fuse implementation).